### PR TITLE
Update LogViewerController@search to be able to match multiple words in query

### PR DESCRIPTION
**Pull request by amacsmith**
Description:
Currently the search function on returns a log if the whole query string is contained in the entry. By exploding the query string and checking if the $value->header contains that strings you are able to search multiple words found anywhere in the $value->header. This allows you to search, say for instance, "(customername) (username) (error_type/exception_class)" and regardless of the order of that query you can find any entry that matches all 3. Basically, it is doing three queries and if each needle of that query is matched then an entry has been matched.

I could fork the could and do this myself. I just thought for me it would be easier and your repo would benefit from its implementation.

Steps To Reproduce:
to add this functionality, update the LogViewerController@search() to look like:

```
public function search(Request $request, $date, $level = 'all')
{
    $query   = $request->get('query');
    
    if (is_null($query))
        return redirect()->route($this->showRoute, [$date]);

    $needles = explode(' ', $query);

    $log     = $this->getLogOrFail($date);
    $levels  = $this->logViewer->levelsNames();
    $entries = $log->entries($level)->filter(function ( LogEntry $value) use ($needles) {

        foreach($needles as $needle){
            if(!Str::contains($value->header, $needle)){
                return false;
            }
        }

        return true;
        
    })->paginate($this->perPage);

    return $this->view('show', compact('level', 'log', 'query', 'levels', 'entries'));
}
```

### CODE REVIEW

File: src/Http/Controllers/LogViewerController.php [hunk 1]
```diff
@@ -141,14 +141,24 @@ public function showByLevel(Request $request, $date, $level)
     public function search(Request $request, $date, $level = 'all')
     {
         $query   = $request->get('query');
-
+
```
Comment (arcanedev-maroc) [inline src/Http/Controllers/LogViewerController.php]: You need to remove the empty spaces

File: src/Http/Controllers/LogViewerController.php [hunk 2]
```diff
@@ -141,14 +141,24 @@ public function showByLevel(Request $request, $date, $level)
...
         if (is_null($query))
             return redirect()->route($this->showRoute, [$date]);
 
+        $needles = explode(' ', $query);
+
         $log     = $this->getLogOrFail($date);
         $levels  = $this->logViewer->levelsNames();
-        $entries = $log->entries($level)->filter(function (LogEntry $value) use ($query) {
-            return Str::contains($value->header, $query);
+        $entries = $log->entries($level)->filter(function ( LogEntry $value) use ($needles) {
```
Comment (arcanedev-maroc) [inline src/Http/Controllers/LogViewerController.php]: Revert the changes to this;
```php
$query   = $request->get('query');

if (is_null($query))
    return redirect()->route($this->showRoute, [$date]);

$log     = $this->getLogOrFail($date);
$levels  = $this->logViewer->levelsNames();
$needles = explode(' ', $query);
$entries = $log->entries($level)->filter(function (LogEntry $value) use ($needles) {
    return Str::contains($value->header, $needles);
})->paginate($this->perPage);

return $this->view('show', compact('level', 'log', 'query', 'levels', 'entries'));
```

**Be careful with the spacing**

**Comment by arcanedev-maroc** [comment]
Can you also add a test for this feature ?

File: src/Http/Controllers/LogViewerController.php (hunk 2 above)
Comment (amacsmith) [reply src/Http/Controllers/LogViewerController.php]: Cool I didn't realize Str::containsAll() existed, good catch.

**Comment by amacsmith** [comment]
> Can you also add a test for this feature ?
> 
> You can add it after this: https://github.com/ARCANEDEV/LogViewer/blob/master/tests/RoutesTest.php#L75-L93

Yep

**Comment by arcanedev-maroc** [comment]
Thanks for the PR :+1:

**Comment by amacsmith** [comment]
Thanks for the cool package!
